src/geo.c

```diff
@@ -527,16 +531,69 @@ void georadiusGeneric(client *c, int flags) {
                        (i+1) < remaining &&
                        !(flags & RADIUS_NOSTORE))
             {
-                storekey = c->argv[base_args+i+1];
+                if (!(flags & GEOSEARCH)) storekey = c->argv[base_args+i+1]; // GEOSEARCH flag don't update storekey
```

Comment:
i would rather return a syntax error if `STORE` present in when GEOSEARCH is set.

p.s. please avoid line comments (`//`)